### ISSUE TITLE
Suggestion for translation project

### ISSUE THREAD

**Issue by SIRSteiner** [yanus171/Handy-News-Reader#144]
What are you thinking about managing translation at OneSky?

https://sirsteiner.oneskyapp.com/collaboration/project?id=154734

**Comment by ThomasLeigh** [comment]
The most important thing is that I'd like to be the only person to handle both Polish and English translation. Could it be done?

**Comment by SIRSteiner** [comment]
No problem. I manage "my" translators ;-) Tomorrow we can try it...

**Comment by yanus171** [comment]
Is it free?

**Comment by yanus171** [comment]
Can we still discuss English strings on github?

**Comment by SIRSteiner** [comment]
OneSky is not free but if you don't need professional translators it doesn't cost anything. There you can find many other apps like meteogram, lithium, musicolet...

**Comment by SIRSteiner** [comment]
... and yes, why not discuss here and translate at OneSky

**Comment by SIRSteiner** [comment]
Another option could be transifex.com but here you can't edit the souce (english) language

**Comment by ThomasLeigh** [comment]
For Me it is not so important where to discuss the translation - what could make a difference is if I could be automatically notified every time some new strings have arrived, and know what they are, and be able to translate them right off the bat. This could be much helpful.

**Comment by SIRSteiner** [comment]
Notifications are not a problem. At the moment I think transifex could be the better choice because you can link it with github. The project https://github.com/barbeau/gpstest also use it. Transifex is free for open source projects.
@yanus171 Do you know this platform?

**Comment by ThomasLeigh** [comment]
But we need to be able to edit the source language as well. Regardless of what language we set as the default one (the source).

**Comment by SIRSteiner** [comment]
The workflow should be:
1. Discuss at GitHub (if needed)
2. Edit source language, commit, pull request at GitHub
3. If configured (via Travis CI) changes in source language resources (at GitHub) will be updated automatically (at Transifex).
4. Translators gets notifications (from Transifex) about changed/new strings.
5. If 100% translated in any language Transifex commits the xml file(s) to GitHub

Transifex also checks syntax, so I found some mistakes in plurals:

> /FlymFork/src/main/res/values-zh/strings.xml
Error while uploading translation: b"Could not import file: Invalid plural types for string: number_of_new_entries. Language supports: ['other'], but found: ['one', 'other'] instead."

> /FlymFork/src/main/res/values-pl/strings.xml
Error while uploading translation: b"Could not import file: Invalid plural types for string: number_of_new_entries. Language supports: ['one', 'few', 'many', 'other'], but found: ['one', 'other'] instead."

> /FlymFork/src/main/res/values-cs/strings.xml
Error while uploading translation: b"Could not import file: Invalid plural types for string: number_of_new_entries. Language supports: ['one', 'few', 'many', 'other'], but found: ['one', 'few', 'other'] instead."

> /FlymFork/src/main/res/values-sk/strings.xml
Error while uploading translation: b"Could not import file: Invalid plural types for string: number_of_new_entries. Language supports: ['one', 'few', 'many', 'other'], but found: ['one', 'few', 'other'] instead."

**Comment by ThomasLeigh** [comment]
For Me https://sirsteiner.oneskyapp.com seems to look good (I would need to test it more deeply to clarify if it is in fact friendly enough - but I need to know what the official choice will be), and - more importantly - it looks comprehensive, non-techy user friendly.

**Comment by SIRSteiner** [comment]
You can also test translation at Transifex
https://www.transifex.com/sirsteiner/handy-news-reader-flymfork/

**Comment by ThomasLeigh** [comment]
What about the access to the English translation?

**Comment by SIRSteiner** [comment]
At Transifex there is no access to the source language. But there are some tipps how to handle this:
https://community.transifex.com/t/can-i-change-the-source-strings-in-transifex/92

**Comment by yanus171** [comment]
Explain me please why are you concerned with such a tools. Our strings updates are not so intensive (last prefs reconfiguration is exclusion). Is it hard to make pull request on github?

**Comment by SIRSteiner** [comment]
Even it is not hard to make pull request for me because I played with GitHub since years, it is much easier managing translations with these tools.

1. You can add new strings to the source language and translators gets notifications.
2. Translators gets notifications if a developer changes a strings to update them.
3. It's easy to add new languages and invite a translator who is not a developer.
4. You and the translators can notice the translation progress.

I help in translation to german language of many project and all of them using any tool (OneSky, Crowdin, Transifex, GetLocalization...). It is much better - believe me.

**Comment by SIRSteiner** [comment]
P.S.: You could add a link to the translation project in Handy to invite translators and add more languages. It's easier to find translators if the number of strings are small. Later it could be difficult if there are (maybe) more than thousand strings.

**Comment by ThomasLeigh** [comment]
If Alexey will give it a chance and if the source language will be changed to some other one in order to have the access to the English translation - I'm willing to give a try.

Is this tool always up-to-date with the source language file from GitHub?

**Comment by SIRSteiner** [comment]
The sync from GitHub to Transifex can be automated with Travis CI. Both (Transifex integration and Travis CI) must configured by Alexey. It is not so easy if he never did it.
If he don't want it OneSky is an easy option. Everytime the source language is changed one of us had to upload the new strings.xml and Alexey can download the translated files everytime he needs for a new release.

**Comment by yanus171** [comment]
@SIRSteiner can you configure the sync and make a pull request?

**Comment by yanus171** [comment]
And a link to translation project can be added by a pull request

**Comment by SIRSteiner** [comment]
@yanus171 I did some tests and it works. But you cannot setup integration with pull requests because transifex and travis ci needs access to your repo. Therefore if you want that it is in your hand you have to do some things. But if you don't want this I can do it for you with my repo and I can make pull requests if needed. What do you think?

**Comment by SIRSteiner** [comment]
P.S.: If I should it do for you then you should send me a notification if you change the source language

**Comment by yanus171** [comment]
And if I will do it in my repo I would not have to notify Travis about string changes?

**Comment by SIRSteiner** [comment]
No, everytime you commit changes the commands in .travis.yml are running. It boots a virtual machine at travis, install transifex client and makes a "tx push -s" to push changes in source language to transifex. You could also push changes in translation that way to transifex but it would make no sense because translators have to put their changes at transifex.

If you don't know how to setup, I can help you step by step.

**Comment by SIRSteiner** [comment]
Which os you are running at your local machine? I hope linux...

**Comment by yanus171** [comment]
Windows :)

**Comment by SIRSteiner** [comment]
I don't know how you work with Windows and local git. Is it similar to linux? Is it also a commandline tool?

**Comment by yanus171** [comment]
Yes, command line tool, where I exec all git * commands. I comes with official git distributive for Windows.

Also Android Studio has git and github integration

**Comment by SIRSteiner** [comment]
Okay we can also start without installing the local transifex client I think. Are you ready to start? Here are the first steps:

Note: please merge my pull request #151 without config files or nothing at all

1. Login at transifex.com with your GitHub-Account and authorize transifex.
1.1 Follow instructions, don't forget to click "Are you translating open source content..." at the bottom of "Company/Organisation" dialog and create a project:
    - Handy News Reader
    - public project
    - open souce
    - URL to GitHub repo
    - file-based project
    - choose source language (en)
    - choose no target languages
    - don't upload anything, we will do it later via travis ci
1.2 Generate a token in user settings - API token and copy/save token (you will need it later and it will be hidden after closing the window)

2. Login at travis-ci.org with your GitHub-Account and authorize travis
2.1 Enable repository at your account settings
2.2 Go to settings of your enabled repository and add the following secret environment variables
    - TRANSIFEX_API_TOKEN with the saved value from 1.2

Next steps are the config files. Tell me if you are ready

**Comment by yanus171** [comment]
I found on Transifex site that there is a free 15 days trial. So is this a paid service?

**Comment by SIRSteiner** [comment]
Please read this:
https://docs.transifex.com/projects/open-source-project
Thats why I wrote 

> don't forget to click "Are you translating open source content..." at the bottom of "Company/Organisation" dialog

**Comment by yanus171** [comment]
I have done your actions. Whats next?

**Comment by SIRSteiner** [comment]
Perfect. Now look at the .tx/config at repo. There you can find two sections
[handy-news-reader.strings-xml]
and
[handy-news-reader.arrays-xml]
If you named the project "Handy News Reader" at transifex you should find "handy-news-reader" in the url like transifex.com/sirsteiner/**handy-news-reader**/dashboard/
If you named the project other than "Handy News Reader" you have to change .tx/config
Do you have to change .tx/config?

**Comment by yanus171** [comment]
I have no need to change config. Url https://www.transifex.com/sirsteiner/handy-news-reader/ shows me language translation percentages. Thank you for help. What next?

**Comment by SIRSteiner** [comment]
No I mean your URL to transifex not sirsteiner

**Comment by SIRSteiner** [comment]
In the meanwhile I updated cs and sk to prevent import error at transifex. Please merge my latest commit

**Comment by SIRSteiner** [comment]
As you can see at #153 there runs the script but something went wrong. But we will find the reason.
What is the url to your transifex project?

**Comment by yanus171** [comment]
1. As I wrote https://www.transifex.com/sirsteiner/handy-news-reader/
2. Yes I have added that var

**Comment by yanus171** [comment]
Sorry, url is https://www.transifex.com/a-250/handy-news-reader-1/dashboard/

**Comment by SIRSteiner** [comment]
Okay, you have to change the two section headings .tx/config
[handy-news-reader.strings-xml]
and
[handy-news-reader.arrays-xml]

to

[handy-news-reader-1.strings-xml]
and
[handy-news-reader-1.arrays-xml]

**Comment by SIRSteiner** [comment]
Should I do it for you?

**Comment by yanus171** [comment]
done

**Comment by SIRSteiner** [comment]
Now look here:
https://github.com/yanus171/Flym/pull/153#partial-pull-merging
Let's have a look:
https://travis-ci.org/yanus171/Flym/builds/527015590?utm_source=github_status&utm_medium=notification
Please click at restart build to see if .tx/config the problem was

**Comment by SIRSteiner** [comment]
Now it works:
https://travis-ci.org/yanus171/Flym/builds/527040628
Please look at transifex. There you should see the source language?

**Comment by yanus171** [comment]
I merged pull request manually (I shouldn't do it?) , after that I restarted build and it failed.

**Comment by yanus171** [comment]
Sorry I need to interrupt now till tomorrow

**Comment by SIRSteiner** [comment]
Okay till tomorrow evening

**Comment by SIRSteiner** [comment]
The problem is that everytime you get a pull request travis build starts. You have to disable it in travis settings for flym "Build pushed pull requests":
https://travis-ci.org/yanus171/Flym/settings

Please merge my latest pull request, then you should see all translation files at transifex. If so, remove the last line from .travis.yml and nearly all things are done.

Good night

**Comment by ThomasLeigh** [comment]
Please let Me know when I can use it freely and where exactly.

**Comment by SIRSteiner** [comment]
@ThomasLeigh: At the moment we are working on automatic sync from GitHub to Transifex. We will inform you when everything works
@yanus171: Did you checked the api token and spelling var name?

**Comment by yanus171** [comment]
I don't understand the problem. I can see hidden api token with key TRANSIFEX_API_TOKEN. After last release I have got a mail, that it was build successfully

**Comment by yanus171** [comment]
I have regenerated api and recreated var TRANSIFEX_API_TOKEN

**Comment by SIRSteiner** [comment]
Thank you. Now please change something in the code to check if it works.

**Comment by SIRSteiner** [comment]
Just a small commit (i.e. adding a new unused string) ...

**Comment by SIRSteiner** [comment]
Thank you build is starting....

**Comment by SIRSteiner** [comment]
Look at transifex!

**Comment by SIRSteiner** [comment]
All translation files are uploaded, yes?

**Comment by SIRSteiner** [comment]
tx INFO: Skipping 'es' translation (file: FlymFork/src/main/res/values-es/arrays.xml).
tx INFO: Skipping 'eu' translation (file: FlymFork/src/main/res/values-eu/arrays.xml).
tx INFO: Skipping 'fi' translation (file: FlymFork/src/main/res/values-fi/arrays.xml).
tx INFO: Skipping 'pl' translation (file: FlymFork/src/main/res/values-pl/arrays.xml).

The only files are missing. Let me show...

**Comment by yanus171** [comment]
How can we translate the text in transifex https://www.transifex.com/a-250/handy-news-reader-1/translate/#ru/strings-xml/171392090?q=translated%3Ano ? I can only view it.

**Comment by SIRSteiner** [comment]
Dashboard -> languages -> click on language

**Comment by SIRSteiner** [comment]
Did you received my requests?

**Comment by yanus171** [comment]
I have trnslated two strings to russian. What's next?

**Comment by yanus171** [comment]
What request?

**Comment by SIRSteiner** [comment]
I did a 4 requests to translate at your transifex page. You should receive an email

Next step is to check whether all files are uploaded (strings.xml and arrays.xml)

**Comment by SIRSteiner** [comment]
If I saw it right arrays.xml for es, eu, fi and pl are not uploaded?

**Comment by yanus171** [comment]
I didnt receieve mails. Please write WHERE you want me to check something. In github or transifex?

**Comment by SIRSteiner** [comment]
Please look at transifex

**Comment by yanus171** [comment]
I still dont understand: I have translated some stirngs. Where they should appear ? As new pull request on github? Or merge automatically? Should I push some button after translation? What about possible conflicts?

**Comment by SIRSteiner** [comment]
Please step by step. We didn't configure sync from transifex to github yet

**Comment by yanus171** [comment]
I what we have configured?

**Comment by SIRSteiner** [comment]
At this moment we have configured that everytime you change source string (or others) it will be uploaded via travis to transifex. And we started to upload the actual translation files. They must be complete at transifex

**Comment by yanus171** [comment]
So we welcome @ThomasLeigh to translate files to Poland?

**Comment by SIRSteiner** [comment]
Please don't translate at this moment!

**Comment by SIRSteiner** [comment]
Please again: check at transifex all translation files!

**Comment by yanus171** [comment]
How should I check them?

**Comment by yanus171** [comment]
By the way, I have a message in transifex page: "Hello yanus171, there are 14 days left in your "A" trial"
while I have checked the open source checkbox while created thу project. Is it ok?

**Comment by SIRSteiner** [comment]
At transifex:
1. Dashboard
2. Whithin Handy News Reader click at "Languages"
3. Click at **every** language and see if there is a strings.xml **and** arrays.xml

**Comment by SIRSteiner** [comment]
It seem you forgot to click "Are you translating open source content..." at the bottom of "Company/Organisation". But it is no problem. After the trial period it turns to a open source company. The project itself is checked as open source

**Comment by SIRSteiner** [comment]
Found at transifex:
>Try out the full feature set for free during our 15 day trial, then have your account automatically convert to the open source plan for your qualified open source projects.

**Comment by SIRSteiner** [comment]
Back to our configuration... Some translation files missing at transifex?

**Comment by yanus171** [comment]
Why they should be missing? Is any way to see the hole file tree?

**Comment by SIRSteiner** [comment]
I saw this at travis 1 hour ago:
>tx INFO: Skipping 'es' translation (file: FlymFork/src/main/res/values-es/arrays.xml).
tx INFO: Skipping 'eu' translation (file: FlymFork/src/main/res/values-eu/arrays.xml).
tx INFO: Skipping 'fi' translation (file: FlymFork/src/main/res/values-fi/arrays.xml).
tx INFO: Skipping 'pl' translation (file: FlymFork/src/main/res/values-pl/arrays.xml).

So please check these files. If you don't check it we cannot do the next steps. Please!

**Comment by yanus171** [comment]
It seems that all files exist. 2 file in each language. Why should they be missing? What if we will decide to add or remove some translation files later?

**Comment by yanus171** [comment]
The files exists, but has no translations

**Comment by SIRSteiner** [comment]
Okay, please change the last line in .travis.yml from
 - tx push -t --skip
to
 - tx push -t -f --no-interactive
and commit it

**Comment by yanus171** [comment]
Now only pl, es, eu arrays files has no strings

**Comment by yanus171** [comment]
done

**Comment by SIRSteiner** [comment]
Job at travis is working...

**Comment by SIRSteiner** [comment]
Now all translation files completed?

**Comment by yanus171** [comment]
Yes, now all (100-50%)

**Comment by SIRSteiner** [comment]
Now please remove the last line from .travis.yml we don't need this anymore because future pushes are only neccessary for the source language

**Comment by yanus171** [comment]
done

**Comment by SIRSteiner** [comment]
Now please remove the Test-string  to see if sync to transifex works

**Comment by yanus171** [comment]
done

**Comment by SIRSteiner** [comment]
It's done and at transifex the test1 string should be disappeared. Is it?

**Comment by SIRSteiner** [comment]
It must ;-) Now we are ready to speak about syncing back from transifex to github. You have the choice: commit or pull request. I would prefer pull requests from transifex to github. And you?

**Comment by yanus171** [comment]
And whats the difference?

**Comment by yanus171** [comment]
And I can not find string <item quantity="one">One new article.</item> in source English file

**Comment by yanus171** [comment]
Thank you, SIRSteiner. Sorry for my impationce. Can we continue tomorow?

**Comment by SIRSteiner** [comment]
Yes we can. I search for the reason... Good Night

**Comment by SIRSteiner** [comment]
> And I can not find string "One new article." in source English file

You only see the "other" string and that's right. In translation view you will see all needed plurals.

Back to sync: Commit or pull request... Hm... I don't know what is better to keep in sync with your local repo and get notifications if any translator did some changes. What do you think?

**Note: Please don't translate anything at the moment. At first we have to setup syncing back to github!**

**Comment by SIRSteiner** [comment]
What you should also know: commits or pull requests gets github from transifex only if a language is fully translated. If you need updated but uncompleted translations for a release... let's speak about it later

**Comment by Ryuno-Ki** [comment]
https://github.com/yanus171/Flym/issues/144#issuecomment-488829245

The difference?
I would expect a commit to add changes right away (IIRC Transifex has a review feature for translations).

Whereas with PR you would have the usual features (diff view, triggered tests on TravisCI or similar). I'd expect a similar behaviour like Greenkeeper or Renovate bots (in case you came across them).

So ... a commit would be fully automated, whereas you'd have some manual work to do with PRs.

**Comment by SIRSteiner** [comment]
Thank you @Ryuno-Ki for you explantation. Also you can always switch between commit and pull request. So @yanus171 let's start with commit. Are you ready?

**Comment by SIRSteiner** [comment]
Go to transifex - Handy News Reader - settings. At the right side you should find "Integrations" tab. Click on it. You should see "GitHub" section. Is it linked or unlinked?

**Comment by ThomasLeigh** [comment]
I'd like to kindly remind that I'd like to be able to manage both English and Polish translation files.

**Comment by yanus171** [comment]
I tried to link and end with following screen:

Select files to syncronize
GITHUB
Mapping
The following configurations is used to map files in a GitHub repository/directory to resources in Transifex.
To learn more about it go to our documentation

Paste your YAML config

**Comment by yanus171** [comment]
@ThomasLeigh, I think you can always do it with a pull request on github

**Comment by SIRSteiner** [comment]
@yanus171 sorry I was sleeping. Now I am ready again.
Now we can link...

**Comment by SIRSteiner** [comment]
1. Click "Link Repository"
2. Select repo and branch
3. Click "Next"
4. At "Select files" you need a valid config. Here is it:

 ```
filters:
   - filter_type: file
     source_file: FlymFork/src/main/res/values/strings.xml
     file_format: ANDROID
     source_language: en
     translation_files_expression: FlymFork/src/main/res/values-<lang>/strings.xml
   - filter_type: file
     source_file: FlymFork/src/main/res/values/arrays.xml
     file_format: ANDROID
     source_language: en
     translation_files_expression: FlymFork/src/main/res/values-<lang>/arrays.xml
```
Paste this code into "Paste your YAML config"
Now you should see "Config file is valid" at the bottom. If not, please make a screenshot.
If yes click at "Next"

5. At "Sync Content" scroll down and you can choose between pull request and commit. If choosed click "Save & Sync"

That's all. Now please make a test and change one translation string. Tell me if it works

**Comment by SIRSteiner** [comment]
> @ThomasLeigh, I think you can always do it with a pull request on github

Yes that's right and the right way. But there is another way we could go. Let's talk about it later when sync source language from github to transifex and sync translations from transifex to github are working.

**Comment by yanus171** [comment]
@SIRSteiner, sorry, but today I am on vacation. The connection here is unstable, I will try to do your action when I can.

**Comment by SIRSteiner** [comment]
> One "problem" is that everytime you get a pull request travis build starts. You have to disable it in travis settings for flym "Build pushed pull requests":
> https://travis-ci.org/yanus171/Flym/settings

@yanus171 Did you forgot it?

**Comment by SIRSteiner** [comment]
I assume you linked transifex with github?
If yes please...
1. Go to transifex
2. Click at Teams
3. There you can find some requests at translation team(s) or so. Please confirm every translators you know

**Comment by yanus171** [comment]
done

**Comment by SIRSteiner** [comment]
Hooray! Now please merge my pull request (mark untranslated strings) and don't forget to change flym-settings at travis ci

**Comment by ThomasLeigh** [comment]
For Me using this solution make sense only if it liberates Me from _GitHub_, so I'd love to have all key features related both to the Polish and English translation.

**Comment by yanus171** [comment]
I have setup setting

**Comment by SIRSteiner** [comment]
Don't forget, there is another way....
Please wait til we have done the setup completely


> > @ThomasLeigh, I think you can always do it with a pull request on github
> 
> Yes that's right and the right way. But there is another way we could go. Let's talk about it later when sync source language from github to transifex and sync translations from transifex to github are working.

**Comment by SIRSteiner** [comment]
@yanus171 Now we have only one problem, there are all strings.xml and arrays.xml twice. I will look at...

**Comment by ThomasLeigh** [comment]
@SIRSteiner I remember that You've pointed out possible solutions for this, I'm just not sure how will it be going, it sounds so complicated. It amazes Me how all those things are not average user-friendly to establish. If those things (what You discuss now, and _Weblate_ and _Jekyll_) would be constructed with the average user kept in mind in the first place - much more people would benefit out of it. But now, even if those things are "open" or "free software" - it doesn't matter as much, because they still are available mostly (if not only) for those who have much more knowledge on "know-how". My dream is to make all of these highly valuable things really available for the general public.

I know this is not the place for this matter, I've just... feel sad about it. _GNU/Linux_... only Steve Jobs has handled to make it. This world would be so much better if the _GNU/Linux_ would be tailored to fit average users perspective. I know that it doesn't seem possible, i.a. thanks to _Windows_, business, all that stuff... but some things could be done.

**Comment by SIRSteiner** [comment]
@ThomasLeigh It seems complicated because I cannot help with windows. Thats why we have to do the setup manually and it's difficulter. The translators will not see what is behind the tools. They will only see the transifex page. But please let's talk about it later.

**Comment by SIRSteiner** [comment]
@yanus171 Please commit a small test with a new string at the source translation

**Comment by SIRSteiner** [comment]
I think I found the problem. @yanus171 Did you linked github at transifex with pull request?

**Comment by Ryuno-Ki** [comment]
@ThomasLeigh I get your point.

> I'm just not sure how will it be going, it sounds so complicated.

Part of this is how the developers behind those services designed the interaction (= APIs). It's hard to imagine, with what people will come up!

If you feel motivated enough, I'd encourage you to write Transifex (and/or GitHub) a mail with your experience. That could help them make the service better, so the next one may not take so many hops.

> It amazes Me how all those things are not average user-friendly to establish.

Historically, they were written from developers to developers.
Recently (= last couple of years) we realised, we need to open up.
But it's so rare to have designers as part of a OSS team. (This point could fill several pages - feel free to take the discussion elsewhere).

> constructed with the average user kept in mind in the first place - much more people would benefit out of it

Sadly, the „average user” (whoever that might be) doesn't speak up. So his opinion can't influence the design.
Plus, as I sad before, without people with knowledge in design, you won't find user surveys.

> But now, even if those things are "open" or "free software" - it doesn't matter as much, because they still are available mostly (if not only) for those who have much more knowledge on "know-how".

Well, I could say much words about this as well. But it's not the right thread (=> flag it as unrelated, so it gets collapsed in GitHub?)

Only so much: There are people building services on top of OSS. With the income, the software should be improved.

Otherwise, it's stuck to work of volunteers (like in case of Jekyll. GitHub pages use it under the hood btw).

> Steve Jobs has handled to make it. This world would be so much better if the GNU/Linux would be tailored to fit average users perspective.

Have you read about his story with Apple (i.e. he was thrown out before)?

Today, Apple software is a Golden Cage: it works wonders - if you stay within its ecosystem. Outside of it less so.

>  but some things could be done

Please fill issues, write about it or talk about your struggles. We can't read your mind ;-)

**Comment by yanus171** [comment]
@SIRSteiner I have the following in Sync options:


Sync content options
GITHUB
PULL CONTENT
Fetch content automatically
A background task will look for new changes in your repos and import to Transifex
PUSH TRANSLATIONS
When would you like Transifex to push content?
The integration will make one commit per language
* 100% translated
100% reviewed
 100% proofread

**Comment by yanus171** [comment]
There is no commit /pull request options now. Only the only variant Push content automatically

**Comment by SIRSteiner** [comment]
@yanus171, @ThomasLeigh Sorry I don't know why it doesn't work as expected. Here it works but I did my setup via local transifex client. I don't want to give up but maybe the other option (I manage translations at OneSky and make pull request) would be the better choise. Then you can spend your time with the code and Thomas is happy because he can edit english and polish language at the same time. And setup is done in 5 minutes

**Comment by ThomasLeigh** [comment]
@SIRSteiner I am amazed by how much of Your persistence and will You've invested so far. I am very, very grateful for that because I'd love to see this world full of beings of a similar attitude. Just let Me know what You will decide.

**Comment by SIRSteiner** [comment]
Thank you @ThomasLeigh !
I spend this time because I find it very useful (if it works). But with transifex and manual configuration I also see problems. That's why I think about an alternative.

**Comment by SIRSteiner** [comment]
@ThomasLeigh Before we start another setup would you do a test with me? Is it possible to chat?

**Comment by ThomasLeigh** [comment]
@SIRSteiner Yes, sure. If You mean a real-time chat, however, I can't predict when exactly, but I'm sure we can handle it.

**Comment by SIRSteiner** [comment]
I quickly setup a chat at: https://www.beisteiners.de/chat Please login without password

**Comment by SIRSteiner** [comment]
@ThomasLeigh and @yanus171 If you have some minutes this evening please come to the chat

**Comment by ThomasLeigh** [comment]
@yanus171  Alexey, we have a great chat party, come enjoy Us :) !

**Comment by yanus171** [comment]
You'd better stay it a bit earlier...

**Comment by SIRSteiner** [comment]
Sorry, you are in another time zone. We did some tests with Crowdin. Pro: Integration much easier, notifications at new or changed strings. At the moment I see no contra. Today I am online after 10.00pm (time zone Berlin). I think you are sleeping then...

**Comment by yanus171** [comment]
So, let's try. What should I do? May be pull requests from you will be sufficient?

**Comment by yanus171** [comment]
I have setup crowdin project https://crowdin.com/project/handy-news-reader/settings#integration Can you create config file?

**Comment by yanus171** [comment]
Can I remove integration with Travis / Transifex?

**Comment by SIRSteiner** [comment]
@yanus171 First: yes you can remove travis (.travis.yml) and transifex (.tx directory). Second: Please remove all translations and source language from crowdin. They will be synced after github integration. Third: Did you request an open source project to get the open source license at Crowdin?

**Comment by SIRSteiner** [comment]
@yanus171 I saw you didn't upload any translation files. So it is okay you don't have to remove anything

**Comment by SIRSteiner** [comment]
Yesterday I wrote a little manual from the beginning. But you are at point 6 I think if you always did an open source request. If not you should do it now.

1. Login at Crowdin with your GitHub account and authorize Crowdin
2. Send an open source project request at https://crowdin.com/page/open-source-project-setup-request.
3. If Crowdin checked that it is open source you get an email. They invite you to create the project.
4. Create the project at Crowdin and select source language and translations.
5. After creation send an email reply with the url to this project. Then Crowdin should grand you the open source license.
6. At Crowdin: go to settings of your project "Integrations". Click at GitHub and "Set up integration".<br>
6.1. Select repository and branch. You should see a red exclamation mark at the right side. Click on the pen to edit.<br>
6.2 Leave config file name untouched and click "Continue"<br>
6.3 Add file filter:
     - Source file path = /FlymFork/src/main/res/values/strings.xml
     - Translated files path = /FlymFork/src/main/res/values-%two_letters_code%/strings.xml
     - Translate attributes: off
     - Content segmentation: off

     Click the green "Save" button at the top of the window<br>
6.4 Click "Add file filter" at the top of the window
     - Source file path = /FlymFork/src/main/res/values/arrays.xml
     - Translated files path = /FlymFork/src/main/res/values-%two_letters_code%/arrays.xml
     - Translate attributes: off
     - Content segmentation: off

     Click the green "Save" button at the top of the window<br>
6.5 Click "Save" at the bottom of the window<br>
6.6 Click "Save" at the bottom of the window. Sync should start.<br>

**Comment by SIRSteiner** [comment]
@yanus171 and @ThomasLeigh Are you ready for a small chat?

**Comment by SIRSteiner** [comment]
@yanus171 I think you have to merge the pull request https://github.com/yanus171/Flym/pull/163 manually because at this time there was travis enabled and it ended with an error.

**Comment by yanus171** [comment]
After I close crowdin PR without merge, sync give me another new PR with 
all these non existing languages. Dispite that I have removed them from 
translating in crowdin. How can get rid of them?

**Comment by SIRSteiner** [comment]
Please look at my comment: https://github.com/yanus171/Flym/pull/167#issuecomment-490613389

**Comment by SIRSteiner** [comment]
Did you see it? It's not a big problem, you should only start integration from zero. Don't forget to remove PR after removing integration and before you restart setup integration

**Comment by SIRSteiner** [comment]
@yanus171 It seems that everything works. Now merge the PR from l10n_master2 and delete branch l10n_master (the old PR branch from crowdin) then we are in sync ;-)
At the moment every translator is able to help in translation. If you want to invite translators you have to change this setting.

**Comment by ThomasLeigh** [comment]
Please let Me know, when and where I should go to do English and Polish translations.

**Comment by ThomasLeigh** [comment]
> At the moment every translator is able to help in translation without invitation. If you want to invite translators you have to change this setting.

Could I have the honor to be the exclusive and the only one translator for English and Polish?

**Comment by SIRSteiner** [comment]
@ThomasLeigh Now Alexey have to merge the pull requests and after that we have to check it (i.e. build apk). If it will be ok then we are ready for the translation process. To get exclusive rights for translations Alexey must change setting for invitation.
Also we have to discuss how we will handle source language (maybe within the chat this weekend?)

**Comment by yanus171** [comment]
@SIRSteiner I see problem with last crowdin PR (#168) :in every language the absent strings are pasted with English text. So we can lose the untranslated strings

**Comment by yanus171** [comment]
@ThomasLeigh I am afraid of words forever and exclusive in free open-source project.

**Comment by SIRSteiner** [comment]
@yanus171 Thats only the half truth. Look at crowdin (the future central for translation). Here you can see that these strings are untranslated. Crowdin replaces untranslated strings with the source like you will see them in the app

**Comment by SIRSteiner** [comment]
@yanus171 There is also an option in settings at the crowdin project. Look at Export: "Skip untranslated strings". Maybe this could help

**Comment by SIRSteiner** [comment]
https://support.crowdin.com/advanced-project-setup/#export

**Comment by yanus171** [comment]
done.

**Comment by yanus171** [comment]
Now in New PR I see that there are few news strings. Why? And the order is changed so it's hardly to check the strings. Can you examine - de strings - is it OK?

**Comment by yanus171** [comment]
I have got a courage to merge it. At first glance it seems ok. See https://github.com/yanus171/Flym/releases/tag/v0.8.2

**Comment by SIRSteiner** [comment]
@yanus171 I think also it is okay. Your questions: 1. Where do you see a few new strings? 2. The order of all translations should be like the source language. That's why at the beginning you had many commits. You can always change the order of source language.

**Comment by SIRSteiner** [comment]
@yanus171 Building apk with the latest translation files works - so I think we are ready to speak about handling source language. @ThomasLeigh In my opinion you can try to translate your home language.

**Comment by yanus171** [comment]
What should we do with source language?
С уважением, Алексей

SIRSteiner <notifications@github.com> 10 мая 2019 г. 16:18:27 написал:

**Comment by SIRSteiner** [comment]
Thomas want to edit source language at crowdin. This is not possible. Source language can only get an update via GitHub. But you can use an unused (and maybe never used) translation (such as esperanto) to do it. Thomas could commit his changes via this translation at crowdin, then you get a PR, merge it and after all you have to copy these translation file(s) to the source language. Could this be a way?

**Comment by yanus171** [comment]
Editing source language is not attended to translation process. It could be easily done in github.

**Comment by SIRSteiner** [comment]
Okay. Let's start with it. My translation were merged as expected, I think Thomas can also start with it.
If his translations also merged successfully then we can add a link at help screen in the app. Are you aggree with it?

**Comment by ThomasLeigh** [comment]
Please allow to do English translation through _Crowdin_ whatever way. It could be significantly helpful to have all the _Crowdin_ benefits regarding both translations. For Me this is a huge difference.

**Comment by yanus171** [comment]
Of cause

**Comment by SIRSteiner** [comment]
As I descriped at we cannot edit source strings in the direct way at crowdin. So we have to use another language code and everytime Thomas did some changes Alexey has to copy that xml file to the source destination manually. Not a problem but Alexey must agree. We only must choose a language that is maybe never needed.

**Comment by ThomasLeigh** [comment]
1. Can we create a fictional language for this purpose - or must to choose out of those which already exist?
2. Will it be working both ways? Will I be notified if any new English strings arrived?

**Comment by SIRSteiner** [comment]
1. It must be real but unused language such as esperanto
2. Everytime Alexey adds a new string or changes a string all translator will get a notification. Alexey don't edit any translation, only source.

**Comment by SIRSteiner** [comment]
The only things Alexey has to do is adding another target language at crowdin and copy value/strings.xml and arrays.xml to the new "fictional" directory values-XX (XX stands for the language code).

**Comment by SIRSteiner** [comment]
There are two possible workflows:
1. You always "translate" **all** strings. Then Alexey has only copy the whole files from values-XX to the source.
2. You only change strings that is needed. Then Alexey has to look at the commit and only change that strings in the source manually

**Comment by yanus171** [comment]
And what if I don't agree with this commit? I need to ask Thomas to reverse 
strings back? And to check that he did it correctly? And to wait for him to 
do it if I need to modify English myself (for new features)? And what if 
someone else would like to edit English while he don't know about 
Esperanto? And what if he wants to translate to Esperanto?

Too many questions which are absent in github pull request.

**Comment by Ryuno-Ki** [comment]
> As I descriped at we cannot edit source strings in the direct way at crowdin. So we have to use another language code and everytime Thomas did some changes Alexey has to copy that xml file to the source destination manually. Not a problem but Alexey must agree. We only must choose a language that is maybe never needed.

Crowdin supports Klingon (Not kidding!).
That's the popular language used by Chaos Computer Club IIRC (albeit they're not using Crowdin AFAIK).

**Comment by ThomasLeigh** [comment]
I don't know all the technical stuff behind the scene - what matters to Me is that I had a hope that it could be much better. _GitHub_ isn't even close to be user-friendly when to comes to translation process, being up-to-date with all new strings, and so on. I can use it but I was excited it could be finally done in an appropriate way, making life easier rather than harder.

But anyway, I am significantly grateful that You were willing to give it a chance - and that other translators will be able to benefit from it in a wider scale.

**Comment by yanus171** [comment]
English is not for translation

**Comment by ThomasLeigh** [comment]
I mean it is for improvement.

**Comment by SIRSteiner** [comment]
@ThomasLeigh, @yanus171: I understand you both. While Alexey  sees the problems from the view of a developer, Thomas sees it from the view of an spell check man for the english language. Over all it is very important that the source language files match the source code. Therefore it is better to discuss necessary changes at the source language here or at crowdin or via separate pull requests despite GitHub is not that easy for everyone.

But back to Crowdin at all: I think now we have found an useful platform and maybe we can find more translators for other languages. @yanus171 Alexey please think about that with the actual setting everyone who has an acount at crowdin can help or also can bother translation. I have had both at other projects. That's why I prefer invitations for translation to have a better control over translators.

**Comment by ThomasLeigh** [comment]
Could You please point out which address I should use for Polish translation? Should I login exactly the same place as before while testing with You, Ronny?

**Comment by SIRSteiner** [comment]
https://crowdin.com/project/handy-news-reader is the right place

**Comment by yanus171** [comment]
@SIRSteiner , Last Crowdin PR reased  <string name="about_us_content"> why?

**Comment by yanus171** [comment]
@SIRSteiner I see no crowds of people willing to translate yet. Lets restrict access a bit later, when these crowds will appear

**Comment by ThomasLeigh** [comment]
@SIRSteiner When we've tested it, I need to synchronize the translation by hand through settings. I can't find this option now - maybe I don't need to synchronize the translation after I've done?

**Comment by SIRSteiner** [comment]
@yanus171 I saw the problem of lost translations. I searched and found this:
https://support.crowdin.com/configuration-file/#changed-strings-update
Please edit the crowdin.yml and add the following option in both file sections:
update_option: update_as_unapproved
Sorry I didn't know it. 

@ThomasLeigh You don't need a manual sync

**Comment by yanus171** [comment]
@SIRSteiner, done. But I don't understand the meaning of setting

**Comment by SIRSteiner** [comment]
@yanus171 At Crowdin members can have different roles.
1. Manager = you
2. Proofreader
3. Translator = Thomas and I
4. Blocked (not a really role...)

Part of the workflow of translation is:

1. Translators translate and they can vote translations (if more translations exists)
2. Proofreader can translate and **approve** translations. Approved translations means that they are correct.

The option "update_option: update_as_unapproved" means that if source string was changed the target translations are marked as unapproved if they were approved before. Then translators get a notification that a string was changed to check the translation.

Therefore it would be better to change the role of Thomas and me to "Proofreader". Then we can mark all correct translated strings of our supported languages to "approved" and everytime you change a source string we get a notification to verify the associated translation.

**Comment by yanus171** [comment]
Thank you for explanation. But how can Thomas approve the transaltion to **other** languages which he doesnt know?

**Comment by yanus171** [comment]
@SIRSteiner, can you edit about app string to add a right link to help with translation?

**Comment by SIRSteiner** [comment]
@yanus171 Done.
@ThomasLeigh You did small changes in the source language. That's okay ;-) But please remember: If you use an apostroph or quotation mark you must "escape" it with a back slash. For example:
> Notifications on new articles and the app's behavior

must be

> Notifications on new articles and the app\\'s behavior

If you forget it the build of the apk fails

**Comment by SIRSteiner** [comment]
@yanus171 I did many changes at the about page. Please look if all links are wanted. Thank you

**Comment by ThomasLeigh** [comment]
@SIRSteiner Thank You for pointing this out. Should I correct these strings then?

**Comment by SIRSteiner** [comment]
@ThomasLeigh No, I already did it ,-)

> Should the apostrophe be preceded by a single backslash?
Yes, "escape" a character means that.

**Comment by ThomasLeigh** [comment]
Thank You.

**Comment by SIRSteiner** [comment]
> ...and everytime you change a source string we get a notification to verify the associated translation.

@ThomasLeigh I changed one source string (about_us_content). Did you got a notification? If not I don't know why. But you will see it if you are looking at Crowdin.

**Comment by ThomasLeigh** [comment]
@SIRSteiner 
I have those:

> "May 10, 2019 17:20
> 1 string was added to the project Handy news reader"
> 
> "Yesterday:
> 20 strings were added to the project Handy news reader"

**Comment by SIRSteiner** [comment]
@ThomasLeigh I mean today. I updated one string, it was set as "unapproved" at Crowdin but I didn't got a notification.

**Comment by ThomasLeigh** [comment]
@SIRSteiner Neither I.

**Comment by SIRSteiner** [comment]
This string was changed:
https://crowdin.com/translate/handy-news-reader/14/en-pl#724

**Comment by yanus171** [comment]
I have merged Thomas pull request. May be now you will get notification?

**Comment by SIRSteiner** [comment]
This is not what I mean. Alexey you saw that I completely changed the about content string at source language. You merged it and Crowdin source language was updated. So far so good. Then I expected a notification that one source string was changed / marked as unapproved.

**Comment by SIRSteiner** [comment]
BTW: @ThomasLeigh you changed the line:
> You can help in translation to your own language at <b><a href=\\"https://crowdin.com/project/handy-news-reader\">Crowdin</a></b>.<br/>

to

> You can help in translation to your own language at <b><a href=\\"https://crowdin.com/project/handy-news-reader">Crowdin</a></b>.<br/>

You removed the back slash. Remember: quotation marks must be "escaped". Please add it again. Thank you

**Comment by SIRSteiner** [comment]
... that's why @yanus171 don't want to open editing of the source ;-)

**Comment by ThomasLeigh** [comment]
What do You mean?

**Comment by SIRSteiner** [comment]
> You removed the back slash. Remember: quotation marks must be "escaped". Please add it again.

I must correct myself: The problem was that you removed only one backslash in this line. I will consistently remove all backslashes at quotation marks...

**Comment by SIRSteiner** [comment]
@yanus171 and @ThomasLeigh: What do you think? Now translation at Crowdin works in my opinion fine. Can we close this issue and start a new one for discussion about source language and new ones for translations only if needed?

**Comment by ThomasLeigh** [comment]
@SIRSteiner 

> What do You think?

What exactly You'd like to know regarding My Own opinion?

**Comment by SIRSteiner** [comment]
Close or not ;-) I vote for close because the real issue is solved.

@yanus171 You can merge my PR, I merged your changes to my repo **before** I did the changes. There should be no problems

**Comment by ThomasLeigh** [comment]
I don't mind to close it.

**Comment by yanus171** [comment]
Last translation commit again has unexpected strings deletions
С уважением, Алексей

SIRSteiner <notifications@github.com> 12 мая 2019 г. 19:50:46 написал:

**Comment by ThomasLeigh** [comment]
Where I should put comments on translation if I'd need to consult anything, regardless of Polish or English translation? I'd like Alexey to know wherever I need to consult something.

**Comment by comradekingu** [comment]
@SIRSteiner Could you add this project on Hosted Weblate. I am not a fan of the tracking and spying Crowdin does, and I don't want my work to enter their closed Translation memory. :)

**Comment by SIRSteiner** [comment]
@comradekingu We choosed Crowdin because of the GitHub integration and free for open source projects. Why should we switch to Weblate? Do Weblate have GitHub integration to keep translations in sync with the source code?

**Comment by comradekingu** [comment]
@SIRSteiner Weblate also has GitHub integration. https://docs.weblate.org/en/latest/vcs.html?highlight=github#github 
As opposed to Crowdin it is actually gratis for libre software.
With Crowdin, no source of revenue is allowed to comply with their terms of doing so, and one has to contribute to their closed source TM when doing so. It would be illegal to call that "gratis" where I am from.

**Comment by ThomasLeigh** [comment]
I have at least 3-year of a web server subscription therefore I could help. Although I am not so into technical matters and probably would need some help with installation and GitHub connection.

**Comment by comradekingu** [comment]
@ThomasLeigh Weblate has a guy hired that does that sort of thing.
Android on GitHub is standard fare, just need to decide exactly what you want once set up.
I can help with that if added as an admin.

**Comment by ThomasLeigh** [comment]
I have no competence to take this decision, I’d like just to help in case @SIRSteiner would go in that direction.
